data/downloads/recursive-table-download.py

```diff
@@ -0,0 +1,100 @@
+import requests
+from bs4 import BeautifulSoup
+import os
+from urllib.parse import urljoin, urlparse
+
+# This script downloads the files from the CEDA archive for the ESA CCI River Discharge dataset.
+# The script navigates recursively through the directory structure to find the download links and save the files.
+
+# Base URL
+base_url = "https://data.ceda.ac.uk/neodc/esacci/river_discharge/data/WL/v1.1/CSV/merged_timeseries/"
+
+# Directory to save the downloaded files
+download_dir = "download/xco2"
+
+# Boolean flag to track if any failures occurred
+had_failures = False
+
+# Function to log messages to a file
+def log_message(message):
+    print(message)
+
+# Recursive function to navigate through the links and download the file
+def find_and_download(url):
+    try:
+        # Fetch the page content
+        response = requests.get(url)
+        response.raise_for_status()
+    except requests.RequestException as e:
+        log_message(f"Failed to fetch {url}: {e}")
+        global had_failures
+        had_failures = True
+        return
+
+    soup = BeautifulSoup(response.content, "html.parser")
+    # Find the table containing the links
+    # This works because the tables containing the links have a the same specific class
+    table = soup.find("table", class_="table table-sm")
```

Comment:
Interesting approach 🤔 Maybe it's easier and less error-prone if we use FTP within a shell script for downloading:
ftp://anon-ftp.ceda.ac.uk/neodc/esacci/river_discharge/data/WL/v1.1/CSV/merged_timeseries

I should have told you before that most datasets are also available via FTP.